raiden/network/transport/matrix/transport.py

```diff
@@ -550,24 +553,35 @@ def send_global(self, room: str, message: Message) -> None:
 
         assert self._global_rooms.get(room_name), f'Unknown global room: {room_name!r}'
 
-        def _send_global():
-            text = JSONSerializer.serialize(message)
-            room = self._global_rooms[room_name]
-            self.log.debug(
-                'Send global',
-                room_name=room_name,
-                room=room,
-                data=text.replace('\n', '\\n'),
-            )
-            room.send_text(text)
-
-        greenlet = self._spawn(_send_global)
-        greenlet.name = (
-            f'MatrixTransport.send_global '
-            f'node:{pex(self._raiden_service.address)} '
-            f'room:{room} '
-            f'user_id:{self._user_id}'
-        )
+        self._global_send_queue.put((room_name, message))
+
+    def _global_send_worker(self):
+        while True:
+            if self._stop_event.is_set():
+                break
+            if self._global_send_queue.qsize() > 0:
+                room_name, message = self._global_send_queue.get()
+
+                def _send_global(room_name, message):
+                    text = JSONSerializer.serialize(message)
+                    room = self._global_rooms[room_name]
+                    self.log.debug(
+                        'Send global',
+                        room_name=room_name,
+                        room=room,
+                        data=text.replace('\n', '\\n'),
+                    )
+                    room.send_text(text)
```

Comment:
If you have a queue of messages, which can have multiple messages on it, you can send them batched, which is more efficient, as it cost a single http roundtrip. Just `'\n'.join` them. You may need to have separate queues though.